modules/acknowledgements/jsx/acknowledgementsIndex.js

```diff
@@ -0,0 +1,370 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+import swal from 'sweetalert2';
+import Modal from 'Modal';
+import Panel from 'Panel';
+import Loader from 'Loader';
+import FilterableDataTable from 'FilterableDataTable';
+
+/**
+ * Acknowledgements Module page.
+ *
+ * Serves as an entry-point to the module, rendering the whole React
+ * component page on load.
+ *
+ * Renders main page consisting of FilterableDataTable, CitationPolicy
+ * and addAcknowledgement.
+ *
+ * @author Zaliqa Rosli
+ * @version 1.0.0
+ *
+ */
+class AcknowledgementsIndex extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      error: false,
+      isLoaded: false,
+      affiliationsOptions: {
+        douglas: 'Douglas',
+        mcgill: 'McGill',
+      },
+      degreesOptions: {
+        bachelors: 'Bachelors',
+        masters: 'Masters',
+        phd: 'PhD',
+        postdoc: 'Postdoctoral',
+        md: 'MD',
+        registeredNurse: 'Registered Nurse',
+      },
+      rolesOptions: {
+        investigators: 'Investigators',
+        projectAdministration: 'Project Administration',
+        databaseManagement: 'Database Management',
+        interviewDataCollection: 'Interview Data Collection',
+        dataAnalyses: 'Data Analyses',
+        mriAcquisition: 'MRI Acquisition',
+        dataEntry: 'Data Entry',
+        databaseProgramming: 'Database Programming',
+        imagingProcessingAndEvaluation: 'Imaging Processing and Evaluation',
+        geneticAnalysisAndBiochemicalAssays: 'Genetic Analysis and Biochemical Assays',
+        randomizationAndPharmacyAllocation: 'Randomization and Pharmacy Allocation',
+        consultants: 'Consultants',
+        lpCsfCollection: 'LP/CSF Collection',
+      },
+      showModal: false,
+    };
+
+    this.fetchData = this.fetchData.bind(this);
+    this.pickElements = this.pickElements.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+    this.openModalForm = this.openModalForm.bind(this);
+    this.closeModalForm = this.closeModalForm.bind(this);
+    this.renderCitationPolicy = this.renderCitationPolicy.bind(this);
+    this.renderAddForm = this.renderAddForm.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   * Additionally add hiddenHeaders to global loris variable
+   * for easy access by columnFormatter.
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
```

Comment:
Should we remove console.error?

Reply:
im not sure. @HenriRabalais should we be?

Reply:
Not really sure. It's more a useful too for testing to have this hear, but as @maltheism pointed it, our jsx linter doesn't like it. I don't really have an opinion either way.

Reply:
i think the jsx linter on major was changed? the console logs for .error passes on the branch (or has been in previously merged PRs) so i think i'll keep it!